### PR TITLE
chore(reserve): sample data improvements

### PR DESCRIPTION
**Pull request by vladopajic**
### Checklist

- [x] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md).
- [ ] My change requires a documentation update, and I have done it.
- [x] I have added tests to cover my changes.
- [ ] I have filled out the description and linked the related issues.

### Description

These changes are needed by PR https://github.com/ethersphere/bee/pull/4040. They are made in individual PR in order to prevent future merge conflicts on https://github.com/ethersphere/bee/pull/4040.

### CODE REVIEW

**Review by mrekucci — COMMENTED** [review]
The [TestAgent test](https://github.com/ethersphere/bee/actions/runs/4891901054/jobs/8733357585?pr=4049) is failing.

**Comment by vladopajic** [comment]
@mrekucci tests should be fixed now, please review again.

File: pkg/storageincentives/proof.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+package storageincentives
+
+import (
+	"github.com/ethersphere/bee/pkg/cac"
+	storer "github.com/ethersphere/bee/pkg/storer"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+func sampleChunk(items []storer.SampleItem) (swarm.Chunk, error) {
```
Comment (mrekucci) [inline pkg/storageincentives/proof.go:13]: Would be nice to have a general description of the `sampleChunk` and `sampleHash` functions even they are not exported.

File: pkg/storer/reserve.go [hunk 2]
```diff
@@ -7,20 +7,20 @@ package storer
 import (
```
Comment (mrekucci) [inline pkg/storer/reserve.go:7]: The "reserveLock" constant is not used and can be removed.

**Review by mrekucci — APPROVED** [review]

**Pull request merged**